ironfish/src/network/messageRegistry.ts

```diff
@@ -97,22 +88,6 @@ const parseRpcNetworkMessage = (
   }
 }
 
-const parseGossipNetworkMessage = (
-  type: NetworkMessageType,
-  bodyWithHeader: Buffer,
-): GossipNetworkMessage => {
-  const { nonce, remaining: body } = GossipNetworkMessage.deserializeHeader(bodyWithHeader)
-
-  switch (type) {
-    case NetworkMessageType.NewBlock:
-      return NewBlockMessage.deserialize(body, nonce)
-    case NetworkMessageType.NewTransaction:
-      return NewTransactionMessage.deserialize(body, nonce)
-    default:
-      throw new Error(`Unknown gossip network message type: ${type}`)
-  }
-}
```

Comment:
It's not necessarily related to this PR so I'd be okay having it elsewhere, but it did make me think it'd be nice to have a test that sends a buffer with a nonexistent message from a Peer to ensure it doesn't crash somewhere in the PeerNetwork code -- it should just disconnect the Peer.

Reply:
I looked into this a bit and it would require a lot of mocking. We'd have to mock the NodeDataChannel in WebRTCConnection. We could just add a test in `WebRTCConnection` but that wouldn't really test that the node doesn't crash. I can add a task for this though

Reply:
Ah I see, I found the try/catches that handle invalid messages in the WebRTCConnection and WebSocketConnection. It seems clear it won't crash from invalid messages, so I don't think we imminently need a test for this.

If someone ends up writing a test, I'd imagine creating a PeerNetwork, creating a Peer with a mocked websocket connection (we already mock `ws` in the PeerNetwork tests), then calling the `socket.onmessage` function on the mocked websocket connection directly and make sure an exception doesn't happen anywhere.